com.unity.shadergraph/Editor/Drawing/Controllers/BlackboardController.cs

```diff
@@ -212,62 +408,48 @@ void InitializeViewModel(bool useDropdowns)
+                    ViewModel.propertyNameToAddActionMap.Add(name, AddShaderInputAction.AddPropertyAction(shaderInputType));
                 }
                 else
-                    ViewModel.propertyNameToAddActionMap.Add(name, new AddShaderInputAction() { blackboardItemType = shaderInputType, addInputActionType = AddShaderInputAction.AddActionSource.AddMenu });
+                    ViewModel.propertyNameToAddActionMap.Add(name, AddShaderInputAction.AddPropertyAction(shaderInputType));
             }
 
             // Default Keywords next
-            ViewModel.defaultKeywordNameToAddActionMap.Add("Boolean", new AddShaderInputAction() { shaderInputReferenceGetter = () => new ShaderKeyword(KeywordType.Boolean), addInputActionType = AddShaderInputAction.AddActionSource.AddMenu });
-            ViewModel.defaultKeywordNameToAddActionMap.Add("Enum", new AddShaderInputAction() { shaderInputReferenceGetter = () => new ShaderKeyword(KeywordType.Enum), addInputActionType = AddShaderInputAction.AddActionSource.AddMenu });
-            // Built-In Keywords last
+            order.isKeyword = true;
+            order.keywordType = KeywordType.Boolean;
+            ViewModel.defaultKeywordNameToAddActionMap.Add("Boolean", AddShaderInputAction.AddKeywordAction(order));
+            order.keywordType = KeywordType.Enum;
+            ViewModel.defaultKeywordNameToAddActionMap.Add("Enum", AddShaderInputAction.AddKeywordAction(order));
+
+            // Built-In Keywords after that
             foreach (var builtinKeywordDescriptor in KeywordUtil.GetBuiltinKeywordDescriptors())
             {
                 var keyword = ShaderKeyword.CreateBuiltInKeyword(builtinKeywordDescriptor);
-                // Do not allow user to add built-in keywords that conflict with user-made keywords that have the same reference name
-                if (Model.keywords.Any(x => x.referenceName == keyword.referenceName))
+                // Do not allow user to add built-in keywords that conflict with user-made keywords that have the same reference name or display name
+                if (Model.keywords.Any(x => x.referenceName == keyword.referenceName || x.displayName == keyword.displayName))
                 {
                     ViewModel.disabledKeywordNameList.Add(keyword.displayName);
                 }
                 else
                 {
-                    ViewModel.builtInKeywordNameToAddActionMap.Add(keyword.displayName,  new AddShaderInputAction() { shaderInputReference = keyword.Copy(), addInputActionType = AddShaderInputAction.AddActionSource.AddMenu });
+                    order.builtInKeyword = (ShaderKeyword)keyword.Copy();
+                    ViewModel.builtInKeywordNameToAddActionMap.Add(keyword.displayName, AddShaderInputAction.AddKeywordAction(order));
                 }
             }
 
             if (useDropdowns)
```

Comment:
As @russoh pointed out we'll need to handle the dropdowns same as the others, using `BlackboardShaderInputOrder`, missed it during the merge, good catch